### PR TITLE
changed fetch method to add candidates

### LINKED ISSUE #282: Use global fetch method in Board Page
Who: As a developer
What: I want to use the global fetch method in Board Page.
Why:  So the fetches we use in Board Page, can send a user token.

### PR DESCRIPTION
**Pull request by NatsGt**
# Description

Replaced postData fetch method with global fetchFromDb

Fixes #282 

# How to test?

Add candidates on a board

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have followed the name conventions for CSS Classnames and filenames, Components names and filenames, Style filenames, if you are in doubt check the the project README.MD and here https://github.com/HackYourFuture-CPH/curriculum/blob/master/review/review-checklist.md
- [x] I have commented my code, particularly in hard-to-understand areas, if you code was simple enough mark the box anyway
- [x] I have made corresponding changes to the documentation, if you code was simple enough mark the box anyway
- [x] This PR is ready to be merged and not breaking any other functionality

### CODE REVIEW

**Review by Midnighter — COMMENTED** [review]
We're using `async` almost everywhere now. I think it'd be better to standardize on that and remove the `Promise` resolution here. Also, any error will be caught at the error boundary anyway, right? Or do you want to catch it here?

**Comment by senner007** [comment]
> We're using `async` almost everywhere now. I think it'd be better to standardize on that and remove the `Promise` resolution here. Also, any error will be caught at the error boundary anyway, right? Or do you want to catch it here?

Please note that using async/await syntax also returns Promise objects. 
The good thing about using async/await is that it is ubiquitous across languages even though the implementation underneath differs slightly.

**Comment by NatsGt** [comment]
Made a change to return data from fetch method since the response.json throws an error if the method is not get.

File: src/client/containers/fetchMethod/fetch.js [hunk 1]
```diff
@@ -24,6 +24,9 @@ export async function fetchFromDb(endpoint, fetchMethod, postBody = {}) {
   if (!response.ok) {
     throw new ApiError(response.statusText, response.status);
   }
-  const dbData = await response.json();
-  return dbData;
+  if (fetchMethod === 'get') {
```
Comment (senner007) [inline src/client/containers/fetchMethod/fetch.js:27]: You can check the headers to see what it returns
` response.headers.get("content-type");`

File: src/client/containers/fetchMethod/fetch.js [hunk 2]
```diff
@@ -24,6 +24,9 @@ export async function fetchFromDb(endpoint, fetchMethod, postBody = {}) {
...
+    const dbData = await response.json();
+    return dbData;
+  }
+  return response;
```
Comment (Midnighter) [inline src/client/containers/fetchMethod/fetch.js:31]: Hmm, this probably makes sense for now but in general also other requests may return data and sometimes it is interesting to access the response object in GET requests... You're sure this doesn't break any existing code?

File: src/client/containers/fetchMethod/fetch.js (hunk 1 above)
Comment (NatsGt) [reply src/client/containers/fetchMethod/fetch.js:27]: This returns null If is post, put, or delete. if is get it returns "application/json; charset=utf-8"

File: src/client/containers/fetchMethod/fetch.js (hunk 2 above)
Comment (NatsGt) [reply src/client/containers/fetchMethod/fetch.js:31]: I could just return the response for all the methods if someone wants to access any other data of the response object. I tried it with get, post, put and delete and it seems to work fine, cant said I tried it with all the endpoints still.

Comment (Midnighter) [reply src/client/containers/fetchMethod/fetch.js:31]: And it's compatible with all the current containers that use `fetchFromDb`?

Comment (NatsGt) [reply src/client/containers/fetchMethod/fetch.js:31]: for now yes, since we just started implementing it

**Review by Midnighter — APPROVED** [review]

File: src/client/containers/fetchMethod/fetch.js (hunk 2 above)
Comment (NatsGt) [reply src/client/containers/fetchMethod/fetch.js:31]: So just to be sure, should I return the response or the response.json() ?

**Pull request merged**